robolectric/src/main/java/org/robolectric/util/ActivityController.java

```diff
@@ -178,6 +183,14 @@ public void run() {
       }
     });
 
+    ViewRootImpl root = component.getWindow().getDecorView().getViewRootImpl();
+    Display display = Shadow.newInstanceOf(Display.class);
```

Comment:
Could we avoid the shadow and do new Display(new DisplayInfo(), new DisplayAdjustments()... etc)

Perhaps since these are hidden classes anyway we are better just creating the display like you are doing here.

Looks fine.